docs/guide/compatibility-with-browser-features.md

```diff
@@ -0,0 +1,23 @@
+# Compatibility With Browser Features
+
+## Content Security Policy (CSP)
+
+To deploy CSP, certain directives or configs must be set due to Vite's internals.
+
+### [`'nonce-{RANDOM}'`](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Security-Policy/Sources#nonce-base64-value)
+
+When [`html.cspNonce`](/config/shared-options#html-cspnonce) is set, Vite adds a nonce attribute with the specified value to the output script tag and link tag for stylesheets. Note that Vite will not add a nonce attribute to other tags, such as `<style>`. Additionally, when this option is set, Vite will inject a meta tag (`<meta property="csp-nonce" nonce="PLACEHOLDER" />`).
+
+The nonce value of a meta tag with `property="csp-nonce"` will be used by Vite whenever necessary during both dev and after build.
+
+:::warning
+Ensure that you replace the placeholder with a unique value for each request. This is important to prevent bypassing a resource's policy, which can otherwise be easily done.
+:::
+
+### [`data:`](<https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Security-Policy/Sources#scheme-source:~:text=schemes%20(not%20recommended).-,data%3A,-Allows%20data%3A>)
+
+By default, during build, Vite inlines small assets as data URIs. Allowing `data:` for related directives (e.g. [`img-src`](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Security-Policy/img-src), [`font-src`](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Security-Policy/font-src)), or, disabling it by setting [`build.assetsInlineLimit: 0`](/config/build-options#build-assetsinlinelimit) is necessary.
+
+:::warning
+Do not allow `data:` for [`script-src`](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Security-Policy/script-src). It will allow injection of arbitrary scripts.
+:::
```

Comment:
Maybe we could add this as a section of the [Building for Production](https://vitejs.dev/guide/build.html) guide for now. And once we have more than one thing to add to a Compatibility With Browser Features guide, we can move it out.

Reply:
I was thinking of adding about import maps in the same page. But I can move it to that page for now.

Reply:
Ah, that is a good point. It is fine then, let's leave it as a separate page 👍🏼

Reply:
Could we move this to `features.md` instead too? The page also talks about browser features like WebAssembly and Web Workers, and perhaps CSP (and importmaps) also belongs there.

Reply:
I'm fine with that, too. But I feel the `features.md` is already quite long 😅. We can split the page later if we want to though.

Reply:
I think it is somewhat neat that all the features are on a single page 😄 Otherwise splitting may have some pages shorter and have to navigate around.

Reply:
I also vote to go with the features page or build guide for now. Maybe we could do a review of the structure later on and break it if we feel it is too much. I'm worried too that we have too many guides in the sidebar and it is starting to be more difficult to consume.

Reply:
OK. I'll move it to features page👍